File: update/Part 1 - Data Preprocessing/Section 2 -------------------- Part 1 - Data Preprocessing --------------------/Python/plantilla_preprocesado.py
# -*- coding: utf-8 -*-

# Plantilla de Pre Procesado
# Cómo importar librerías
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importar el dataset
dataset = pd.read_csv("Data.csv")
x = dataset.iloc[:, :-1].values
y = dataset.iloc[:, -1].values

# Tratamiento de NAs
from sklearn.impute import SimpleImputer

imputer = SimpleImputer(missing_values=np.nan, strategy="mean")
imputer.fit(x[:, 1:3])
x[:, 1:3] = imputer.transform(x[:, 1:3])

# Codificar datos categóricos
from sklearn.preprocessing import OneHotEncoder, LabelEncoder
from sklearn.compose import ColumnTransformer

columntransformer = ColumnTransformer(
    [("one_hot_encoder", OneHotEncoder(categories="auto"), [0])],
    remainder="passthrough"
    )

x = np.array(columntransformer.fit_transform(x), dtype=np.float_)

labelencoder_y = LabelEncoder()
y = labelencoder_y.fit_transform(y)

# Dividir dataset en entrenamiento y testing
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, 
                                                    test_size=0.2, 
                                                    random_state=0)

# Escalado de variables
"""
from sklearn.preprocessing import StandardScaler
sc_x = StandardScaler()

# En caso de no querer escalar los dummys, se pueden seleccionar 
# solo las columnas que se desean estandarizar.
x_train[:, 3:-1] = sc_x.fit_transform(x_train[:, 3:-1])
x_test[:, 3:-1] = sc_x.transform(x_test[:, 3:-1])
"""

File: update/Part 1 - Data Preprocessing/Section 2 -------------------- Part 1 - Data Preprocessing --------------------/R/plantilla_preprocesado.R
# Plantilla para precosado de datos
# Importar el dataset
dataset = read.csv("Data.csv")

# Tratamiento de NAs
dataset$Age = ifelse(is.na(dataset$Age), 
                     ave(dataset$Age, FUN = function(x) mean(x, na.rm = TRUE)), 
                     dataset$Age)

dataset$Salary = ifelse(is.na(dataset$Salary), 
                     ave(dataset$Salary, FUN = function(x) mean(x, na.rm = TRUE)), 
                     dataset$Salary)
# Codificar datos categóricos
dataset$Country = factor(dataset$Country,
                         levels = c("France", "Spain", "Germany"),
                         labels = c(1, 2, 3))
dataset$Purchased = factor(dataset$Purchased,
                           levels = c("No", "Yes"),
                           labels = c(0, 1))

# Dividir dataset en entrenamiento y testing
# install.packages("caTools")
# library(caTools)

set.seed(123)
# Crear vector que divide los datos de entrenamiento (True) y test (False)
split =  sample.split(dataset$Country, SplitRatio = 0.8)

training_set = subset(dataset, split == TRUE)
testing_set = subset(dataset, split == FALSE)

# Escalado de valores

# training_set[, 2:3] = scale(training_set[, 2:3]) #Solo Edad y Salario
# testing_set[, 2:3] =  scale(testing_set[, 2:3]) # Pais y Comprado no son numericos (son factores)












